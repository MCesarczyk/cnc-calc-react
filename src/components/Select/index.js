import styled from "styled-components";

export const Select = styled.select`
    min-width: 226px;
    border: solid 1px ${({ theme }) => theme.color.primary};
    border-radius: 5px;
    padding: 5px;
`;