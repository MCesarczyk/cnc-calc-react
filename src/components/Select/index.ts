import styled from 'styled-components';

export const Select = styled.select`
  min-width: 16rem;
  border: solid 1px ${({ theme }) => theme.color.primary};
  border-radius: 5px;
  padding: 5px;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    width: 100%;
  }
`;
