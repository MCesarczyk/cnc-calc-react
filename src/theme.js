export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}