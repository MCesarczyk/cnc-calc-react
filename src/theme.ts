export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
