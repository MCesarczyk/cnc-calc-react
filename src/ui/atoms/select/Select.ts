import { CSSProperties } from 'react';
import styled from 'styled-components';

export type SelectVariants = 'DEFAULT';

type SelectStyles = Required<
  Pick<CSSProperties, 'color' | 'backgroundColor' | 'borderColor'> & {
    disabledText?: CSSProperties['color'];
    disabledBackgroundColor?: CSSProperties['backgroundColor'];
  }
>;

export type Palette<TVariants extends string, TStyles> = {
  [variant in TVariants]: TStyles;
};

const styles: Palette<SelectVariants, SelectStyles> = {
  DEFAULT: {
    color: 'primaryText',
    backgroundColor: 'elementBackground',
    borderColor: 'primary',
    disabledText: 'disabledText',
    disabledBackgroundColor: 'disabledBackground',
  },
};

interface SelectProps {
  $variant?: SelectVariants;
  disabled?: boolean;
}

export const Select = styled.select<SelectProps>`
  min-width: 16rem;
  border: solid 1px ${({ theme }) => theme.color.primary};
  border-radius: 5px;
  padding: 5px;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    width: 100%;
  }

  &:disabled {
    background-color: ${({ theme }) => theme.color[styles.DEFAULT.disabledBackgroundColor]};
    color: ${({ theme }) => theme.color[styles.DEFAULT.disabledText]};
    border-color: ${({ theme }) => theme.color[styles.DEFAULT.disabledText]};
  }
`;

Select.defaultProps = {
  $variant: 'DEFAULT',
};

Select.displayName = 'Select';
