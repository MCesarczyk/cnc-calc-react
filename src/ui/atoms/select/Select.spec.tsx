import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { theme } from 'theme';
import { Select } from './Select';

test('Select component rendering properly', () => {
  const select = renderer.create(
    <ThemeProvider theme={theme}>
      <Select />
    </ThemeProvider>,
  );

  let tree = select.toJSON();

  expect(tree).toMatchSnapshot();
});
