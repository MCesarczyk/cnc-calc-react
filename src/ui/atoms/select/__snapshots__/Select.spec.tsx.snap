// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component rendering properly 1`] = `
.c0 {
  min-width: 16rem;
  border: solid 1px #1e3246;
  border-radius: 5px;
  padding: 5px;
}

.c0:disabled {
  background-color: #ddd;
  color: #808080;
  border-color: #808080;
}

@media (max-width: 480px) {
  .c0 {
    width: 100%;
  }
}

<select
  className="c0"
/>
`;
